SimTracker/TrackAssociation/plugins/DigiSimLinkPruner.cc

```diff
@@ -0,0 +1,161 @@
+// -*- C++ -*-
+//
+// Package:    SimTracker/TrackAssociation
+// Class:      DigiSimLinkPruner
+//
+/**\class DigiSimLinkPruner DigiSimLinkPruner.cc SimTracker/TrackAssociation/plugins/DigiSimLinkPruner.cc
+
+ Description: Produce a pruned version of the DigiSimLinks collection based on the association to a collection of TrackingParticles
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Enrico Lusiani
+//         Created:  Fri, 14 May 2021 08:46:10 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+#include "SimDataFormats/TrackerDigiSimLink/interface/StripDigiSimLink.h"
+
+#include <unordered_set>
+
+namespace {
+
+  template <typename T>
+  std::vector<edm::DetSet<T>> pruneByTpAssoc(const edm::DetSetVector<T>& simLinkColl,
+                                             const std::set<std::pair<uint32_t, EncodedEventId>>& selectedIds) {
```

Comment:
Can you remind why std::set is used here, while std::unordered_set in master?
https://github.com/cms-sw/cmssw/blob/master/SimTracker/TrackAssociation/plugins/DigiSimLinkPruner.cc#L45

Reply:
The ability to use std::unordered_set comes from #33462 which added the UniqueSimTrackId type and its hasher that enables std::unordered_set. It is not available on 11_2_X as far as I know. I had to do the same modification when backporting to 10_6_X.
TrackerHitAssociation/plugins/ClusterTPAssociationProducer.cc for example shows the same difference between 11_2_X and master